src/nerve/server/engine.py

```
@@ -223,55 +227,85 @@ async def _create_node(self, params: dict[str, Any]) -> dict[str, Any]:
         history = params.get("history", True)  # Enable history by default
         response_timeout = params.get("response_timeout", 1800.0)
         ready_timeout = params.get("ready_timeout", 60.0)
+        provider_dict = params.get("provider")  # Provider config dict (optional)
+
+        # Handle provider configuration and start proxy if needed
+        proxy_url: str | None = None
+        if provider_dict is not None:
+            if backend != "claude-wezterm":
+                raise ValueError("provider config is only supported for claude-wezterm backend")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)